internal/registration/internal/application/register_new_user_test.go

```diff
@@ -0,0 +1,95 @@
+package application_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"GoEdu/internal/registration/internal/application"
+	"GoEdu/internal/registration/internal/domain"
+)
+
+var _ application.UserRegistrationRepository = (*RegisterNewUserRepoMock)(nil)
+
+type RegisterNewUserRepoMock struct {
+	added bool
+}
+
+func (r *RegisterNewUserRepoMock) Add(_ context.Context, _ *domain.UserRegistration) error {
+	r.added = true
+	return nil
+}
+func (r RegisterNewUserRepoMock) Load(
+	_ context.Context,
+	_ domain.UserRegistrationID) (*domain.UserRegistration, error) {
+	return &domain.UserRegistration{}, nil
+}
+func (r RegisterNewUserRepoMock) Update(_ context.Context, _ *domain.UserRegistration) error {
+	return nil
+}
+
+var _ application.UniqueEmailVerifier = (*UniqueEmailVerifierSpy)(nil)
+
+type UniqueEmailVerifierSpy struct {
+	checked bool
+}
+
+func (u *UniqueEmailVerifierSpy) IsUnique(_ context.Context, _ domain.UserRegistrationEmail) error {
+	u.checked = true
+	return nil
+}
+
+var _ application.PasswordHasher = (*PasswordHasherSpy)(nil)
+
+type PasswordHasherSpy struct {
+	hashed bool
+}
+
+func (p *PasswordHasherSpy) Hash(password domain.UserPassword) (domain.HashedUserPassword, error) {
+	p.hashed = true
+	return domain.CreateHashedUserPassword(password), nil
+}
+
+func TestPositiveRegisterNewUserRegistration(t *testing.T) {
+	tests := map[string]struct {
+		verifier   *UniqueEmailVerifierSpy
+		hasher     *PasswordHasherSpy
+		repository *RegisterNewUserRepoMock
+		command    application.RegisterNewUserCommand
+	}{
+		"case 1": {
+			verifier:   &UniqueEmailVerifierSpy{},
+			hasher:     &PasswordHasherSpy{},
+			repository: &RegisterNewUserRepoMock{},
+			command: application.RegisterNewUserCommand{
+				FirstName: "a",
+				LastName:  "a",
+				Email:     domain.UserRegistrationEmail{},
+				Password:  domain.UserPassword{},
+			},
+		},
+		"case 2": {
+			verifier:   &UniqueEmailVerifierSpy{},
+			hasher:     &PasswordHasherSpy{},
+			repository: &RegisterNewUserRepoMock{},
+			command: application.RegisterNewUserCommand{
+				FirstName: "Abc",
+				LastName:  "Dfg",
+				Email:     domain.UserRegistrationEmail{},
```

Comment:
Use real VO, not an empty structs. You can create a valid VO in this line with a Must constructors.